packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -692,3 +697,60 @@ class TestHeader extends SliverPersistentHeaderDelegate {
   @override
   bool shouldRebuild(TestHeader oldDelegate) => false;
 }
+
+class _TestLayoutExtentIsNegative extends StatelessWidget {
+
+  const _TestLayoutExtentIsNegative(this.widgetCountBeforeSliverOverlapAbsorber);
+
+  final int widgetCountBeforeSliverOverlapAbsorber;
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Test'),
+        ),
+        body: NestedScrollView(
+          headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
+            return <Widget>[
+              ...List<Widget>.generate(widgetCountBeforeSliverOverlapAbsorber, (_) => SliverToBoxAdapter(
+                child: Container(
+                  color: Colors.red,
+                  height: 200,
+                  margin:const EdgeInsets.all(20),
+                ),
+              ),
+              ),
+              SliverOverlapAbsorber(
```

Comment:
Is there an actual real-world use case for having a bunch of widget come before the SliverOverlapAbsorber?

Reply:
I ran into this issue because I had TabBar & TabBarView at the bottom of my page for displaying different categories of goods, and many other widgets on top of them(e.g., horizontal ListView, AD banner, etc.)